### PR TITLE
Convert SLAM components to Svelte

### PR DESCRIPTION
**Pull request by micheal-parks**
Part of [this migration epic](https://viam.atlassian.net/browse/APP-1995). This PR converts SLAM components from Vue to Svelte, simplifies a lot of business logic in the process, and moves a lot of API code into the `api` folder. All features tested locally, all working as normal.
Screenshot 2023-06-07 at 13 51 12
Screenshot 2023-06-07 at 13 51 49

### CODE REVIEW

File: web/frontend/src/api/slam.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    } else if (end.code !== 0) {
+      const error = {
+        message: end.details,
+        code: end.code,
+        metadata: end.metadata,
+      };
+      reject(error);
+    }
+    const arr = concatArrayU8(chunks);
+    resolve(arr);
```
Comment (mcous) [inline web/frontend/src/api/slam.ts:59]: When wrapping event emitters in `new Promise`, should we be removing the listeners on `resolve`/`reject`? I usually find myself paranoid about memory leaks and setting up a `cleanup` function

File: web/frontend/src/api/motion.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        rcLogConditionally(req);
+        client.robotService.cancelOperation(req, new grpc.Metadata(), (error, response) => (
+          error ? reject(error) : resolve(response)
+        ));
+        return;
+      }
+    }
+
+    reject(new Error('Operation not found!'));
+  });
```
[the remark this answers is not in the document]
Comment (micheal-parks) [reply web/frontend/src/api/motion.ts:73]: This is only canceling one op, hence the early return if a match is found. If it's easier for you to read with `filter()` instead I can update :)

File: web/frontend/src/api/slam.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  });
+});
+
+export const getSLAMPosition = (
+  client: Client,
+  name: string
+) => new Promise<commonApi.Pose | undefined>((resolve, reject) => {
+  const request = new slamApi.GetPositionRequest();
+  request.setName(name);
```
[the remark this answers is not in the document]
Comment (micheal-parks) [reply web/frontend/src/api/slam.ts:68]: Yeah I'm down, will update

File: web/frontend/src/api/slam.ts (hunk 1 above)
Comment (micheal-parks) [reply web/frontend/src/api/slam.ts]: I'm up for being judicious about cleanup, but these GRPC streams don't have any `off`, `unbind`, etc. methods :(

File: web/frontend/src/api/slam.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (micheal-parks) [reply web/frontend/src/api/slam.ts:67]: updated!

File: web/frontend/src/api/motion.ts [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+  );
+
+  return new Promise((resolve, reject) => {
+    client.motionService.moveOnMap(request, new grpc.Metadata(), (error, response) => (
+      error ? reject(error) : resolve(response?.getSuccess())
+    ));
+  });
+};
+
+export const stopMoveOnMap = (client: Client, operations: { op: robotApi.Operation.AsObject }[]) => {
```
[the remark this answers is not in the document]
Comment (micheal-parks) [reply web/frontend/src/api/motion.ts:58]: per discussion with cousins: will address later

File: web/frontend/src/api/slam.ts (hunk 1 above)
Comment (mcous) [reply web/frontend/src/api/slam.ts]: 🙈

**Review by mcous — APPROVED** [review]
Pure TS stuff LGTM! Component migrations look correct, too, but I was not able to actually test

**Pull request merged**